meshtastic/stream_interface.py

```
@@ -213,20 +327,28 @@ def __reader(self) -> None:
                     f"Meshtastic serial port disconnected, disconnecting... {ex}"
                 )
+            else:
+                disconnect_source = "stream.close_requested"
         except OSError as ex:
             if (
                 not self._wantExit
             ):  # We might intentionally get an exception during shutdown
+                disconnect_source = "stream.os_error"
                 logger.error(
                     f"Unexpected OSError, terminating meshtastic reader... {ex}"
                 )
```

Ruff (0.15.1)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)